core/tx_pool.go

```diff
@@ -437,6 +447,22 @@ func (pool *TxPool) loop() {
 					queuedEvictionMeter.Mark(int64(len(list)))
 				}
 			}
+			// Evict pending transactions
+			for addr := range pool.pending {
+				// Skip local transactions from the eviction mechanism
+				if pool.locals.contains(addr) {
+					continue
+				}
+				// Any non-locals old enough should be removed
+				if time.Since(pool.beats[addr]) > pool.config.Lifetime {
```

Comment:
This might not work as expected since our senders keep sending new txns which updates `beats`

Reply:
Could it work once `pool.config.AccountPendingLimit` is reached? Then all the new txs will be non-executable/gapped and within `promoteTx()` we check if the queue is full -> so don't update `pool.beats[addr] = time.Now()`

Reply:
This was actually incorrect, because `beats` was cleaned up together with `queue`. Tried to fix in [d4f6fe0](https://github.com/scroll-tech/go-ethereum/pull/1107/commits/d4f6fe086ffff34ae1c597c6c4089dadca519809).